In [39]:
import pandas as pd
import numpy as np
import matplotlib as plt
import seaborn as sns
import sqlite3

In [40]:
bom_df = pd.read_csv("../Data/bom.movie_gross.csv")

In [41]:
bom_df.head()


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [42]:
bom_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [43]:
bom_df.dtypes

title              object
studio             object
domestic_gross    float64
foreign_gross      object
year                int64
dtype: object

In [44]:
bom_df['domestic_gross'] = pd.to_numeric(bom_df['domestic_gross'],errors = 'coerce')

In [45]:
bom_df['foreign_gross'] = pd.to_numeric(bom_df['foreign_gross'],errors = 'coerce')

In [46]:
bom_df.dtypes

title              object
studio             object
domestic_gross    float64
foreign_gross     float64
year                int64
dtype: object

In [47]:
bom_df.duplicated().sum()

0

In [48]:
bom_df.isnull().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1355
year                 0
dtype: int64

In [49]:
studio_null_percent, domestic_null_percent, foreign_null_percent = (5/3387)*100, (28/3387)*100, (1355/3387)*100
studio_null_percent, domestic_null_percent, foreign_null_percent


(0.14762326542663123, 0.8266902863891349, 40.00590493061706)

Since studio and domestic_gross have a very small percentage of missing values we can drop the missing values in them. foreign_gross has a high percent of missing values hence we fill the missig values using mean or median according to skewness.

In [50]:
bom_df = bom_df.dropna(subset = ['studio'])

In [51]:
bom_df = bom_df.dropna(subset = ['domestic_gross'])

In [52]:
median_foreign_gross = bom_df['foreign_gross'].median()
median_foreign_gross

19600000.0

In [53]:
mean_foreign_gross = bom_df['foreign_gross'].mean()
mean_foreign_gross

75979668.67282717

since mean=median, the data is symmetrical hence we can replace with either

In [54]:
bom_df['foreign_gross'] = bom_df['foreign_gross'].fillna(mean_foreign_gross)

In [55]:
bom_df.isnull().sum()

title             0
studio            0
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

In [56]:
bom_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3356 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3356 non-null   object 
 1   studio          3356 non-null   object 
 2   domestic_gross  3356 non-null   float64
 3   foreign_gross   3356 non-null   float64
 4   year            3356 non-null   int64  
dtypes: float64(2), int64(1), object(2)
memory usage: 157.3+ KB


tn.movie_budgets

In [57]:
tm_df = pd.read_csv('../zippedData/tn.movie_budgets.csv.gz')
tm_df

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

     production_budget domestic_gross worldwide_gross  
0         $425,000,000   $760,507,625  $2,776,345,279  
1         $410,600,000   $241,063,875  $1,045,663,875  
2         $350,000,000    $42,762,350    $149,762,350  
3         $330,600,000   $459,005,868  $1,403,013,963  
4         $317,000,000   $620,181,382  $1,316,721,747  
...                ...            ...             ...  
5777            $7,000             $0              $0  
5778            $6,000        $48,482        $240,495  
5779            $5,000         $1,338          $1,338  
5780            $1,400             $0              $0  
5781            $1,100       $181,041        $181,041  

[5782 rows x 6 columns]

In [58]:
import sqlite3

In [59]:
conn = sqlite3.connect("../Data/im.db")


In [60]:
tables = pd.read_sql("SELECT name FROM sqlite_master WHERE type='table';", conn)
tables

name
0   movie_basics
1      directors
2      known_for
3     movie_akas
4  movie_ratings
5        persons
6     principals
7        writers

In [61]:
im_df = pd.read_sql("SELECT * FROM movie_basics LIMIT 5;", conn)
im_df


movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [62]:
ratings = pd.read_sql(
'''
SELECT * 
FROM movie_ratings 
LIMIT 5;
''',conn)
ratings

movie_id  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [63]:
persons = pd.read_sql("SELECT * FROM persons LIMIT 15;", conn)
persons

person_id           primary_name  birth_year death_year  \
0   nm0061671      Mary Ellen Bauder         NaN       None   
1   nm0061865           Joseph Bauer         NaN       None   
2   nm0062070             Bruce Baum         NaN       None   
3   nm0062195           Axel Baumann         NaN       None   
4   nm0062798            Pete Baxter         NaN       None   
5   nm0062879         Ruel S. Bayani         NaN       None   
6   nm0063198                  Bayou         NaN       None   
7   nm0063432          Stevie Be-Zet         NaN       None   
8   nm0063618              Jeff Beal      1963.0       None   
9   nm0063750        Lindsay Beamish         NaN       None   
10  nm0064023          Bryan Beasley         NaN       None   
11  nm0064350     Martine Beauchemin         NaN       None   
12  nm0065847  Michael Frost Beckner      1963.0       None   
13  nm0065919            Will Becton         NaN       None   
14  nm0066093           Ahmad Bedair      1945.0       None   

                                  primary_profession  
0          miscellaneous,production_manager,producer  
1         composer,music_department,sound_department  
2                         miscellaneous,actor,writer  
3   camera_department,cinematographer,art_department  
4   production_designer,art_department,set_decorator  
5          director,production_manager,miscellaneous  
6                                              actor  
7                                composer,soundtrack  
8               composer,music_department,soundtrack  
9                              actress,miscellaneous  
10                          director,producer,writer  
11                       producer,production_manager  
12                     writer,producer,miscellaneous  
13         miscellaneous,editor,editorial_department  
14                               actor,miscellaneous

In [64]:
im_df = pd.read_sql("""SELECT * FROM movie_basics
                    JOIN movie_ratings
                    USING(movie_id);
                    """,conn)
im_df.head(10)


movie_id                       primary_title  \
0  tt0063540                           Sunghursh   
1  tt0066787     One Day Before the Rainy Season   
2  tt0069049          The Other Side of the Wind   
3  tt0069204                     Sabse Bada Sukh   
4  tt0100275            The Wandering Soap Opera   
5  tt0112502                             Bigfoot   
6  tt0137204                     Joe Finds Grace   
7  tt0146592                         Pál Adrienn   
8  tt0154039                So Much for Justice!   
9  tt0159369  Cooper and Hemingway: The True Gen   

                       original_title  start_year  runtime_minutes  \
0                           Sunghursh        2013            175.0   
1                     Ashad Ka Ek Din        2019            114.0   
2          The Other Side of the Wind        2018            122.0   
3                     Sabse Bada Sukh        2018              NaN   
4               La Telenovela Errante        2017             80.0   
5                             Bigfoot        2017              NaN   
6                     Joe Finds Grace        2017             83.0   
7                         Pál Adrienn        2010            136.0   
8                      Oda az igazság        2010            100.0   
9  Cooper and Hemingway: The True Gen        2013            180.0   

                       genres  averagerating  numvotes  
0          Action,Crime,Drama            7.0        77  
1             Biography,Drama            7.2        43  
2                       Drama            6.9      4517  
3                Comedy,Drama            6.1        13  
4        Comedy,Drama,Fantasy            6.5       119  
5             Horror,Thriller            4.1        32  
6  Adventure,Animation,Comedy            8.1       263  
7                       Drama            6.8       451  
8                     History            4.6        64  
9                 Documentary            7.6        53

In [65]:
im_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   movie_id         73856 non-null  object 
 1   primary_title    73856 non-null  object 
 2   original_title   73856 non-null  object 
 3   start_year       73856 non-null  int64  
 4   runtime_minutes  66236 non-null  float64
 5   genres           73052 non-null  object 
 6   averagerating    73856 non-null  float64
 7   numvotes         73856 non-null  int64  
dtypes: float64(2), int64(2), object(4)
memory usage: 4.5+ MB


In [66]:
im_df.duplicated().sum()

0

In [67]:
im_df.isnull().sum()

movie_id              0
primary_title         0
original_title        0
start_year            0
runtime_minutes    7620
genres              804
averagerating         0
numvotes              0
dtype: int64

In [68]:
runtime_null_percent, genres_null_percent = (7620/73856) * 100, (804/73856) * 100
runtime_null_percent,genres_null_percent


(10.317374350086656, 1.0886048526863086)

genres has a very small percentage of missing values (~1.09%) hence its missing values can be dropped. runtime has a roughly large percentage of missing values hence we'll fill the missing values using mean or median depending on skewness.

In [69]:
im_df = im_df.dropna(subset = ['genres'])

In [70]:
mean_runtime = im_df['runtime_minutes'].mean()
mean_runtime

94.7322732805843

In [71]:
median_runtime = im_df['runtime_minutes'].median()
median_runtime


91.0

Since the mean != median the data is not symmetrical meaning mean is affected by outliers. Therefore, we'll use median to fill the missing values in runtime_minutes.

In [72]:
im_df['runtime_minutes'] = im_df['runtime_minutes'].fillna(median_runtime)

In [73]:
im_df.isnull().sum()

movie_id           0
primary_title      0
original_title     0
start_year         0
runtime_minutes    0
genres             0
averagerating      0
numvotes           0
dtype: int64